integration_tests/e2e/confirm-achieve-goal.cy.ts

```diff
@@ -0,0 +1,152 @@
+      cy.wrap(planDetails.oasysAssessmentPk).as('oasysAssessmentPk')
+      cy.openSentencePlan(planDetails.oasysAssessmentPk)
+    })
+  })
+
+  describe('Security', () => {
+    beforeEach(() => {
+      cy.get<PlanType>('@plan').then(plan => {
+        cy.addGoalToPlan(plan.uuid, DataGenerator.generateGoal({})).then(goal => {
+          cy.wrap(goal).as('newGoal')
+          cy.addStepToGoal(goal.uuid, DataGenerator.generateStep({}))
+        })
+
+        planOverview.agreePlan()
+      })
+    })
+
+    it('Should display authorisation error if user does not have READ_WRITE role', () => {
+      cy.get<string>('@oasysAssessmentPk').then(oasysAssessmentPk => {
+        cy.openSentencePlan(oasysAssessmentPk, 'READ_ONLY')
+      })
+
+      cy.get<Goal>('@newGoal').then(goal => {
+        cy.visit(`/confirm-if-achieved/${goal.uuid}`, { failOnStatusCode: false })
+        cy.get('.govuk-body').should('contain', 'You do not have permission to perform this action')
+      })
+
+      cy.checkAccessibility(true, ['scrollable-region-focusable'])
+    })
+  })
+
+  describe('Rendering', () => {
+    beforeEach(() => {
+      cy.get<PlanType>('@plan').then(plan => {
+        cy.addGoalToPlan(plan.uuid, DataGenerator.generateGoal({})).then(goal => {
+          cy.wrap(goal).as('newGoal')
+          cy.addStepToGoal(goal.uuid, DataGenerator.generateStep({}))
+        })
+
+        planOverview.agreePlan()
+      })
+    })
+
+    it('Displays agree plan page correctly on load', () => {
+      cy.get<Goal>('@newGoal').then(goal => {
+        cy.visit(`/confirm-if-achieved/${goal.uuid}`)
+        cy.get('h1').contains('has achieved this goal')
+        cy.get('.govuk-summary-card__title').should('contain', goal.title)
+      })
+      cy.checkAccessibility()
+    })
+  })
+
+  describe('Submission behaviour', () => {
+    beforeEach(() => {
+      cy.get<PlanType>('@plan').then(plan => {
+        cy.addGoalToPlan(plan.uuid, DataGenerator.generateGoal({})).then(goal => {
+          cy.wrap(goal).as('newGoal')
+          cy.addStepToGoal(goal.uuid, DataGenerator.generateStep({}))
+        })
+
+        planOverview.agreePlan()
+        cy.get<Goal>('@newGoal').then(goal => {
+          cy.visit(`/update-goal-steps/${goal.uuid}`)
+
+          cy.get('#step-status-1').select('Completed') // select completed status
+
+          cy.get('button').contains('Save goal and steps').click()
+        })
+      })
+    })
+
+    it('Confirm goal achieved successfully without optional note', () => {
+      achieveGoal.isGoalAchievedRadio('yes')
+      cy.get('button').contains('Save and continue').click()
+      cy.url().should('include', 'plan?type=achieved&status=achieved')
+      cy.get(':nth-child(3) > .moj-sub-navigation__link')
+      cy.get('.moj-sub-navigation__link').eq(2).should('contain', 'Achieved goals (1)')
+      cy.get('.govuk-summary-card').should('contain', 'Marked as achieved on')
+
+      cy.get<Goal>('@newGoal').then(goal => {
```

Comment:
can't we just click the "view details" link in the summary card? probably follows the above a bit more cleanly.

Reply:
Yeah, makes sense to do it that way